src/lib.rs

```diff
@@ -921,56 +942,99 @@ fn parse_headers_iter_uninit<'a, 'b>(
 
     let mut iter = autoshrink.headers.iter_mut();
 
+    macro_rules! maybe_continue_after_obsolete_line_folding {
+        ($bytes:ident, $label:lifetime) => {
+            if config.allow_obsolete_multiline_headers_in_responses {
+                match $bytes.peek() {
+                    None => {
+                        // Next byte may be a space, in which case that header
+                        // is using obsolete line folding, so we may have more
+                        // whitespace to skip after colon.
+                        return Ok(Status::Partial);
+                    }
+                    Some(b' ') | Some(b'\t') => {
+                        // The space will be consumed next iteration.
+                        continue $label;
+                    }
+                    _ => {
+                        // There is another byte after the end of the line,
+                        // but it's not whitespace, so it's probably another
+                        // header or the final line return. This header is thus
+                        // empty.
+                    },
+                }
+            }
+        }
+    }
+
     'headers: loop {
+        macro_rules! continue_with_new_header_on_next_line {
```

Comment:
I like this macro but the name is confusing when viewed in calling contexts. It should emphasize that this yields the given error if `ignore_invalid_headers` is not enabled. To bikeshed, perhaps `error_or_ignore_header_line`?